ui/hooks/useName.ts

```diff
@@ -33,7 +35,10 @@ export function useName(
   const normalizedValue = normalizeValue(value, type);
   const typeVariationKey = getVariationKey(type, chainId);
   const variationKey = variation ?? typeVariationKey;
-  const nameEntry = names[type]?.[normalizedValue]?.[variationKey];
+  const variationsToNameEntries = names[type]?.[normalizedValue] || {};
+  const nameEntry =
```

Comment:
The huge upside to this approach is we only have to define each account identity petname once in the state, but the downside is these can't currently be deleted from the petnames UI since that is currently using the selected chain as the variation when saving.

I don't think it's worth adding more to this ticket, but something worth discussing for a potential additional ticket or future work.

Reply:
Acknowledged. Created https://github.com/orgs/MetaMask/projects/47/views/1?pane=issue&itemId=48025425 for further discussion.